src/test/java/eu/merloteducation/contractorchestrator/ContractStorageServiceTest.java

```diff
@@ -1322,6 +1422,54 @@ void addContractAttachmentsInDraft() throws IOException {
         assertTrue(result.getNegotiation().getAttachments().contains("myFile.pdf"));
     }
 
+    @Test
+    @org.springframework.transaction.annotation.Transactional(propagation = Propagation.NOT_SUPPORTED) // handle transactions manually
+    void addContractAttachmentsInDraftFail() throws StorageClientException {
+        transactionTemplate = new TransactionTemplate(transactionManager);
+
+        doThrow(StorageClientException.class).when(storageClient).pushItem(any(), any(), any(byte[].class));
+
+        DataDeliveryContractTemplate template = new DataDeliveryContractTemplate(dataDeliveryContract, false);
+        template.setServiceContractProvisioning(new DataDeliveryProvisioning()); // reset provisioning
+
+        transactionTemplate.execute(status -> {
+            contractTemplateRepository.save(template);
+            return "foo";
+        });
+
+        ContractTemplate check1 = transactionTemplate.execute(status -> {
+            return contractTemplateRepository.findById(template.getId()).orElse(null);
+        });
+        assertNotNull(check1);
+        assertFalse(check1.getAttachments().contains("myFile.pdf"));
+
+        Exception thrownEx = null;
+        try {
+            transactionTemplate.execute(status -> {
+                try {
+                    contractStorageService.addContractAttachment(template.getId(), new byte[]{}, "myFile.pdf", "authToken");
+                } catch (IOException ex) {
```

Comment:
can this ioexception ever be thrown? it is caught in the service

Reply:
The service method had an unnecessary "throws IOException". This has been dealt with.